omod/src/main/java/org/bahmni/module/bahmniOfflineSync/strategy/SelectiveSyncStrategy.java

```diff
@@ -0,0 +1,304 @@
+        if (addressHierarchyEntry == null) {
+            return;
+        }
+        wardIDs.add(addressHierarchyEntry.getUserGeneratedId());
+        List<AddressHierarchyEntry> childAddressHierarchyEntries = addressHierarchyService.getChildAddressHierarchyEntries(addressHierarchyEntry);
+        updateWardIds(addressHierarchyService, wardIDs, childAddressHierarchyEntries);
+    }
+
+    private void updateWardIds(AddressHierarchyService addressHierarchyService, List<String> wardIDs, List<AddressHierarchyEntry> childAddressHierarchyEntries) {
+        for (AddressHierarchyEntry childAddressHierarchyEntry : childAddressHierarchyEntries) {
+            getAllWardIds(childAddressHierarchyEntry, addressHierarchyService, wardIDs);
+        }
+    }
+
+    private String trim(String content) {
+        content = content.trim();
+        return content.replaceAll("(\\s*,\\s*)", ",");
+    }
+
+    @Override
+    public List<String> getEventCategoriesList() {
+        List<String> eventCategoryList = new ArrayList();
+
+        eventCategoryList.add("patient");
+        eventCategoryList.add("encounter");
+        eventCategoryList.add("addressHierarchy");
+        eventCategoryList.add("parentAddressHierarchy");
+        eventCategoryList.add("offline-concepts");
+        eventCategoryList.add("forms");
+
+        return eventCategoryList;
+    }
+
+
+
+    private List<String> getFiltersForAddressHierarchy(AddressHierarchyEntry addressHierarchyEntry) {
+        List addressHierarchyFilters = new ArrayList();
+        while (addressHierarchyEntry.getParent() != null) {
+            if (addressHierarchyEntry.getUserGeneratedId().length() == 6) {
+                addressHierarchyFilters.add(addressHierarchyEntry.getUserGeneratedId());
+                break;
+            }
+            addressHierarchyEntry = addressHierarchyEntry.getParent();
+        }
+        return addressHierarchyFilters;
+    }
+
+    @Override
+    public List<EventLog> getEventLogsFromEventRecords(List<EventRecord> eventRecords) {
+        List<EventLog> eventLogs = new ArrayList<EventLog>();
+
+        for (EventRecord er : eventRecords) {
+            EventLog eventLog = new EventLog(er.getUuid(),er.getCategory(),er.getTimeStamp(),er.getContents(), null, er.getUuid());
+            String category = er.getCategory();
+            String uuid = getUuidFromURL(er.getContents());
+            String filter = null;
+
+            if (!uuid.isEmpty()) {
+                if (category.equalsIgnoreCase("all-concepts")) {
+                    if (isOfflineConceptEvent(uuid)) {
+                        eventLog.setCategory("offline-concepts");
+                    } else {
+                        eventLog.setCategory("concepts");
+                    }
+                }
+
+                if (category.equalsIgnoreCase("Patient")|| category.equalsIgnoreCase("LabOrderResults")) {
+                   // filter = evaluateFilterForPatient(uuid);
```

Comment:
@gopikrishna-yaramothu not realised before.. i see comments here too. are these required here ?

Reply:
Removed comments @binduak